### PR TITLE
Type definition to unblock credit card form

### PR DESCRIPTION
**Pull request by cnasc**
# Description

This PR adds some type definitions, the lack of which were blocking progress with the Stripe integration.

(verified in my credit card branch, then pulled out here for review)

# Issues

Refs #3762 

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [x] This PR only contains configuration changes (package.json, etc.)
  - [ ] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [ ] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/vendor.d.ts [hunk 1]
```diff
@@ -1,7 +1,50 @@
...
 declare module 'react-jazzicon'
 
 // TODO: Remove this when we have a TS build of unlock-js
 declare module '@unlock-protocol/unlock-js'
+
+type PaymentMethodType = 'card' | 'card_present'
+
+// vastly simplified from the real types, but this is all we should need
+interface PaymentMethodIncomplete {
+  billing_details?: {
```
Comment (cellog) [inline unlock-app/src/vendor.d.ts]: do all of these need to be ?: or are there a subset of possibilities that we expect? maybe we could do a union type of those subsets?

**Comment by cnasc** [comment]
Hm. Tests pass locally but on CI this seems to have somehow wiped out the declaration for jazzicon and unlock-js

File: unlock-app/src/vendor.d.ts (hunk 1 above)
Comment (cnasc) [reply unlock-app/src/vendor.d.ts]: the `billing_details` implementation here was pulled from the Stripe @types file. These are all optional properties on the object in their API (actually, almost everything is an optional object property). We could possibly restrict it to just the subset of properties we expect to have (or care about -- in the end we only need to get the `id` from `PaymentMethod`), but at some point in the not-too-distant future we should be deleting this code and just pulling in from DefinitelyTyped.

**Comment by cnasc** [comment]
My best guess so far is that you can't declare _and_ augment type definitions in the same file. Maybe.

**Pull request merged**